src/core.ts

```diff
@@ -47,8 +48,12 @@ export function activate() {
     init();
   } catch (e) {
     onActivateErrorUnsafe(e);
+    return;
   }
 
+  // Send client information as the first event
+  sendClientInfo();
```

Comment:
Based on my earlier comment, let's see if we can avoid adding another top-level JSON object. Envelope is (or was) supposed to address this.

Reply:
Please see my response to your previous comment

---
In reply to: [144364782](https://github.com/Microsoft/clarity-js/pull/39#discussion_r144364782) [](ancestors = 144364782)

Reply:
Synced offline: We renamed this to 'metadata' and we'd be passing it as a part of the first payload.

First payload will have the following structure:
{
  envelope: IEnvelope;
  events: IEvent[];
  metadata: IImpressionMetadata;
}

Subsequent payloads will just contain envelope and events.

---
In reply to: [144372712](https://github.com/Microsoft/clarity-js/pull/39#discussion_r144372712) [](ancestors = 144372712,144364782)